### DISCUSSION TITLE
Can I remove the default vercel.app URL for a project used on deployment?

### DISCUSSION THREAD

**Discussion by jschuur** [vercel/community#2075] [Help] [answered]
### Summary

I know how to add a custom URL for a deployed Vercel Next.js app, but can you keep it from also assigning the projectname-username.vercel.app based hostname for a deployment?

### Additional information

In my case, I've deployed an instance of Chatbot UI, but I don't want it out there for someone to guess the URL and use my OpenAI API key, since (currently) it doesn't have its own auth.

I've given it a random GUID based one on vercel.app, but the deployment is also still available via the project/username based one (even after I deleted the deployment and redeployed).

**Comment by jschuur** [comment]
There's a [PR](https://github.com/mckaywrigley/chatbot-ui/pull/215) to add Basic Auth to Chatbot UI, which solves this and I know Vercel also offers auth (but not on the hobby tier) .

**Comment by amyegan** [comment, marked as answer]
Hi @jschuur! Aside from implementing an auth solution, as you mentioned, you could reassign the `<my-project>.vercel.app` domain to a junk deployment. To do that, you would assign the `<my-project>.vercel.app` url to an unused git branch. You could also change the branch so it doesn't contain any sensitive code (a static html page, for example), then deploy it to assign the `<my-project>.vercel.app` URL to this new preview deployment.

For an added layer of protection, you could use the free [Vercel Authentication](https://vercel.com/docs/concepts/deployments/deployment-protection#vercel-authentication) feature to prevent anyone else from accessing preview deployments from the project.

**Reply by jschuur** [reply]
> Hi @jschuur! Aside for implementing an auth solution, as you mentioned, you could reassign the `<my-project>.vercel.app` domain to a junk deployment.

Thanks, Amy, some great ideas there. I didn't realize I could reuse the hostname in another deployment (once I've freed it up I assume).
